src/Playwright/Core/Route.cs

```diff
@@ -83,12 +83,19 @@ public Task ContinueAsync(RouteContinueOptions options = default)
         private async Task RaceWithPageCloseAsync(Task task)
         {
             var page = (Page)Request.Frame.Page;
+            if (page == null)
+            {
+                task.IgnoreException();
+                return;
+            }
 
             // When page closes or crashes, we catch any potential rejects from this Route.
             // Note that page could be missing when routing popup's initial request that
             // does not have a Page initialized just yet.
-            // TODO: ignore exceptions
-            await Task.WhenAny(task, page.ClosedOrCrashedTcs.Task).ConfigureAwait(false);
+            if (task != await Task.WhenAny(task.ContinueWith(t => t.Exception.Handle(_ => true), System.Threading.CancellationToken.None, TaskContinuationOptions.OnlyOnFaulted, TaskScheduler.Default), page.ClosedOrCrashedTcs.Task).ConfigureAwait(false))
+            {
+                task.IgnoreException();
```

Comment:
This will ignore the exceptions twice as they are already ignore above with
```c#
task.ContinueWith(t => t.Exception.Handle(_ => true), System.Threading.CancellationToken.None, TaskContinuationOptions.OnlyOnFaulted, TaskScheduler.Default)
```
Also `ContinueWith` will return a new task so the if condition is always true.

Reply:
feel free to follow-up! Thanks.